### PR TITLE
fix: 12456, 12420, 12422 on reconnect modal issues

### LINKED ISSUE #12420: [Bug]: The save button functionality fails in the reconnect Datasource modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The save button functionality fails in the reconnect Datasource modal
image

### Steps To Reproduce

1. Import the application
2. inside the reconnect Datasource modal
3. Add the datasource details 
4. Now click on text button
5. Now click on Save button
6. The user is not switched to the new datasource instead the user stays in the same data source page 

### Public Sample App

_No response_

### Version

release

### LINKED ISSUE #12422: [Bug]: The data source selection is broken in Reconnect Datasource modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The data source selection is broken in Reconnect Datasource modal
https://www.loom.com/share/94e89538c36d4923b6eb5729f46538e2

### Steps To Reproduce

1.  Import the application
2. inside the reconnect Datasource modal
3. Add the datasource details
4. Now click on test button
5. Now click on Save button
6. The user is not switched to the new datasource instead the user stays in the same data source page
7. now manually click on the next datasource 
8. The user is navigated to wrong datasource

### Public Sample App

_No response_

### Version

release

### LINKED ISSUE #12456: [Bug]: Gsheet authentication fails in the reconnect datasource modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The google auth fails when the user tries to authentication the google sheet as a last datasource

[LOOM DEMO](https://www.loom.com/share/1bc6143a7e4e47aeaeb3f0c801fa94b3)
### Steps To Reproduce

1. Import the application which as multiple data source along with google sheet
2. authenticate all the datasource now at the last authenticate the google sheet 
3. The user is navigated to the application instead of authenticate  page

### Public Sample App

_No response_

### Version

release

### LINKED ISSUE #12511: [Bug]: The Overview spelling seems to be wrong in reconnect Modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The Overview spelling seems to be wrong in reconnect Modal
image

### Steps To Reproduce

1. Import the application
2. inside the reconnect Datasource modal
3. the overview spelling is wrong in the reconnect modal text

### Public Sample App

_No response_

### Version

release

### PR DESCRIPTION
**Pull request by haojin111**
## Description

> Updating logic of datasource selecton on reconnect datasource modal
> Fixed issue of latest gsheet authentication issue
> Renames "RECONNECT APPLICATION" to "RECONNECT".

Fixes #12456 
Fixes #12420 
Fixes #12422 
Fixes #12511 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please see linked issue - manually testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/reconnect-modal-issues 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.34 **(0)** | 37.74 **(0)** | 35.94 **(-0.01)** | 56.56 **(0)**
 :red_circle: | app/client/src/selectors/entitiesSelector.ts | 48.97 **(0.12)** | 13.25 **(0)** | 25.16 **(-0.16)** | 45.24 **(0.03)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

**Comment by haojin111** [comment]
https://www.loom.com/share/2d21817ff27b4e2ea666869f0dbee752

File: app/client/src/pages/Editor/gitSync/ReconnectDatasourceModal.tsx [hunk 1]
```diff
@@ -501,8 +514,8 @@ function ReconnectDatasourceModal() {
   });
 
   const shouldShowDBForm =
-    isConfigFetched && !isLoading && !datasouce?.isConfigured;
-  const shouldShowSuccessMessages = datasouce && datasouce.isConfigured;
+    isConfigFetched && !isLoading && !datasource?.isConfigured;
+  const shouldShowSuccessMessages = datasource && datasource.isConfigured;
```
Comment (IAmAnubhavSaini) [inline app/client/src/pages/Editor/gitSync/ReconnectDatasourceModal.tsx:518]: This can be `datasource?.isConfigured`.

**Review by IAmAnubhavSaini — CHANGES_REQUESTED** [review]
There is another task, "Reconnect Application" should be "Reconnect", when we go to datasources.

**Comment by IAmAnubhavSaini** [comment]
We need tests too.

**Comment by IAmAnubhavSaini** [comment]
**Reconnect**

Screenshot 2022-04-05 at 7 58 30 PM
Screenshot 2022-04-05 at 8 00 40 PM

File: app/client/src/pages/Editor/gitSync/ReconnectDatasourceModal.tsx (hunk 1 above)
Comment (haojin111) [reply app/client/src/pages/Editor/gitSync/ReconnectDatasourceModal.tsx]: resolved it! please check!

**Review by IAmAnubhavSaini — DISMISSED** [review]
Since there is no starting point for state and selector testing, and state being 10k lines, we are not going ahead with redux selector testing.

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**